library/network/test/network_service_test.rb

```diff
@@ -39,5 +46,52 @@
 
       subject.RunSystemCtl("wicked", "disable | evil")
     end
+
+    it "raises an exception when no service name is provided" do
+      expect { Yast::NetworkService.RunSystemCtl(nil, "enable") }.to raise_error
+    end
+  end
+
+  describe "#EnableDisableNow" do
+    subject { Yast::NetworkService }
+
+    context "When changing running service" do
+      before(:each) do
+        allow(subject).to receive(:current_name).and_return(:netconfig)
```

Comment:
this before block and allow can be moved up, and then use let(:current_name) in each context

Reply:
well, matter of preference. And I currently don't see any benefit from switching to your case.

Reply:
well, it is a good practice for DRY

Reply:
i prefer to keep tests short and readable ... e.g. to have mocks as close as possible. I don't like to read whole test when searching how a particular value is mocked.

Reply:
The mocked value is closed to the text as it it declared in the context itself.
```
context.. do
  let(:current_name) { :netconfig }
end
```
and in the text there are mainly expectations.

But it is ok as it is